Add note on target temp validation on ClimateEntity (#2299)
home-assistant/developers.home-assistant: blog/2024-09-24-climate-set-temp-validation.md

---
author: G Johansson
authorURL: https://github.com/gjohansson-ST
authorImageURL: https://avatars.githubusercontent.com/u/62932417?v=4
authorTwitter: GJohansson
title: "Additional validation in Climate set temperature"
---

> As of Home Assistant Core 2024.10, we have implemented further validation into the `set_temperature` method. This means that the integrations does not need to implement this validation in their own methods.

Reviewer: ```suggestion
As of Home Assistant Core 2024.10, we have added further validation of the parameters passed to the `set_temperature` method. This means that integrations don't need to implement this validation in their own methods.
```

> Currently it was possible to use the `temperature` attribute when the entity did not set `ClimateEntityFeature.TARGET_TEMPERATURE` or `target_temp_low`/`target_temp_high` when the entity did not set `ClimateEntityFeature.TARGET_TEMPERATURE_RANGE`. This has been deprecated and will raise an error starting from 2025.4.

Author: Implemented also in 2025.10 so I think it was fitting to include that also on this blog post

> Currently it was possible to use the `temperature` attribute when the entity did not set `ClimateEntityFeature.TARGET_TEMPERATURE` or `target_temp_low`/`target_temp_high` when the entity did not set `ClimateEntityFeature.TARGET_TEMPERATURE_RANGE`. This has been deprecated and will raise an error starting from 2025.4.

Reviewer: ```suggestion
Before 2024.10 it was possible to set the `temperature` parameter in the action when the entity did not set `ClimateEntityFeature.TARGET_TEMPERATURE` or `target_temp_low`/`target_temp_high` parameters when the entity did not set `ClimateEntityFeature.TARGET_TEMPERATURE_RANGE`. This has been deprecated and will raise an error starting from 2025.4.
```

> Before 2024.10 it was possible to set the `temperature` parameter in the action when the entity did not set `ClimateEntityFeature.TARGET_TEMPERATURE` or `target_temp_low`/`target_temp_high` parameters when the entity did not set `ClimateEntityFeature.TARGET_TEMPERATURE_RANGE`. This has been deprecated and will raise an error starting from 2025.4.

Reviewer: I'm reading https://github.com/home-assistant/core/pull/125180 again and wonder if we should just raise `ServiceValidationError`. Even though custom integrations may do this incorrectly, we can't know if it's the integration that wants to use an incorrect parameter of it's just the user that passes an incorrect parameter. I think we'll create a lot of noise with integrations if we ask users to report this.

Sorry for changing my mind this late.

Author: Right, I tihnk it's mainly a user-side problem so I agree. I'll make a follow-up PR to change the behavior and then we change the blog post.

Author: Fixed in https://github.com/home-assistant/core/pull/126692
I will update the blog post accordingly

Reviewer: Thanks!
